packages/block-editor/src/utils/transform-styles/transforms/wrap.js

```diff
@@ -3,17 +3,37 @@
  */
 const IS_ROOT_TAG = /^(body|html|:root).*$/;
 
+const IS_BUTTON_TAG = /^(button).*$/;
+
+const IS_INPUT_TAG = /^(input).*$/;
+
 const wrap = ( namespace, ignore = [] ) => ( node ) => {
 	const updateSelector = ( selector ) => {
+		if ( selector.match( IS_BUTTON_TAG ) ) {
+			return selector.replace(
+				IS_BUTTON_TAG,
+				'button:not(.components-button)'
+			);
+		}
+
+		if ( selector.match( IS_INPUT_TAG ) ) {
+			return selector.replace(
+				IS_INPUT_TAG,
+				'input:not(.components-text-control__input):not(.components-placeholder__input):not(.components-form-token-field__input)'
+			);
+		}
```

Comment:
The "wrap" transform is a generic transform, and this is very specific, can we extract it to its own transform instead? and add a comment clarifying that yes, it's a bit hacky to target some specific Gutenberg components but that it's a good contained hack.

Reply:
@youknowriad yes, that makes sense to me, I just updated the code, let me know if this looks good to you and I'll go ahead and update the tests and move them to a separate file too.

Reply:
Yep, that's hat I had in mind, thanks for the update.